### ISSUE TITLE
Too slow for object detecting.

### ISSUE THREAD

**Issue by springkim** [OlafenwaMoses/ImageAI#85]
I tested RetinaNet and Yolov3 on Windows system with GTX 1080.

But, Yolov3 shows around 11 FPS and RetinaNet shows 2FPS.

Can i improve these models speed?

**Comment by OlafenwaMoses** [comment]
Hello, you can use the speed modes provided by ImageAI when loading the model. See sample below. 

.loadModel(detection_speed="fast")

The available options are fast, faster, fastest and flash.

**Comment by springkim** [comment]
Thanks for your answer.

Regards.

**Comment by OlafenwaMoses** [comment]
You are welcome.

**Comment by mamintoosi** [comment]
Dear Moses
How we can load the model with faster option in training?
`from imageai.Detection.Custom import DetectionModelTrainer

trainer = DetectionModelTrainer()
trainer.setModelTypeAsYOLOv3()
trainer.setDataDirectory(data_directory="datasets/"+ds_name)
trainer.setTrainConfig(object_names_array=["object1"], batch_size=8, num_experiments=20, train_from_pretrained_model="pretrained-yolov3.h5")
trainer.trainModel()`

**Comment by ekesdf** [comment]
Hi,
I have looked in to the Custom Yolo implementation,  and it seems the .loadModel function just exists in the CustomObjectDetection or CustomVideoDetection part of imageai. So you can't change the "training" speed, you can only change the prediction speed

**Comment by ekesdf** [comment]
The .loadModel has no key value detection_speed anymore, so it is useless it is not implemented anymore.

**Comment by mamintoosi** [comment]
@ekesdf
Thank you for your answer.
In the above code, for faster training, is it ok to set self.net_h and self.net_w, in Detection/Custom/generator.py/BatchGenerator to some smaller values than default value 416?

**Comment by ekesdf** [comment]
Technically, it would increase the speed. 
you can test by adding this code 
```
from time import process_time


#This at the bottom of your code
print(process_time())
```
The lower the number gets, the faster your prediction.
But I would decrease the size too much bc the smaller the model gets the lower gets the accuracy.

**Comment by ekesdf** [comment]
Instead of editing the Batchgenerator just edit this
Detection>Custom >__init__.py


```
class CustomObjectDetection:

    """
    This is the object detection class for using your custom trained models. It supports your custom trained YOLOv3 model and allows to you to perform object detection in images.
    """

    def __init__(self):
        self.__model_type = ""
        self.__model_path = ""
        self.__model_labels = []
        self.__model_anchors = []
        self.__detection_config_json_path = ""
        self.__input_size = 416
```

I had a 0.04 sec performance increase by just setting it to 384
from 6.727634083 to 6.761348105

**Comment by mamintoosi** [comment]
Tank you,
But trainer is an instance of DetectionModelTrainer not CustomObjectDetection.
0.04 sec is too low, perhaps this time reduction is only due to different runs.

**Comment by ekesdf** [comment]
Oh, so you want faster training and not faster Prediction :man_facepalming:

**Comment by ekesdf** [comment]
@mamintoosi 
The problem is that when you change only the self.net_h and self.net_w values to a lower value, that those values get recalculated every 10 epochs by this function
```
 # get image input size, change every 10 batches
        net_h, net_w = self._get_net_size(idx)
        base_grid_h, base_grid_w = net_h//self.downsample, net_w//self.downsample

def _get_net_size(self, idx):
        if idx % 10 == 0:
            net_size = self.downsample*np.random.randint(self.min_net_size/self.downsample, \
                                                         self.max_net_size/self.downsample+1)

            self.net_h, self.net_w = net_size, net_size
        return self.net_h, self.net_w
```

and I don't know from where it gets called

**Comment by ekesdf** [comment]
@mamintoosi 
You can just set those values static

for  example, like this
```
 def _get_net_size(self, idx):

        return 384,384
```

**Comment by ekesdf** [comment]
But i am still confused bc in your first question it sounded like you want to increase your prediction  speed and now you wnat to implement the same thing into detection :question:

**Comment by mamintoosi** [comment]
Dear @ekesdf 
Thank you for your immediate responses.
I will check get_net_size.

BTW, In first question I mentioned "training":
**How we can load the model with faster option in training?**
